File: Classes/View Controllers/Tabs/Home/SearchAllViewController.m
//
//  SearchAllViewController.m
//  iSub
//
//  Created by Ben Baron on 4/6/11.
//  Copyright 2011 Ben Baron. All rights reserved.
//

#import "SearchAllViewController.h"
#import "SearchSongsViewController.h"
#import "UIViewController+PushViewControllerCustom.h"
#import "Defines.h"
#import "SavedSettings.h"
#import "EX2Kit.h"
#import "Swift.h"

@implementation SearchAllViewController

- (BOOL)shouldAutorotate {
    if (settingsS.isRotationLockEnabled && [UIDevice currentDevice].orientation != UIDeviceOrientationPortrait) {
        return NO;
    }
    
    return YES;
}

#pragma mark - View lifecycle

- (void)viewDidLoad {
    [super viewDidLoad];
	
	self.cellNames = [NSMutableArray arrayWithCapacity:3];
	
	if (self.listOfArtists.count > 0) {
		[self.cellNames addObject:@"Artists"];
	}
	
	if (self.listOfAlbums.count > 0) {
		[self.cellNames addObject:@"Albums"];
	}
	
	if (self.listOfSongs.count > 0) {
		[self.cellNames addObject:@"Songs"];
	}
	
	if (IS_IPAD()) {
		self.view.backgroundColor = ISMSiPadBackgroundColor;
	}
    
    self.tableView.rowHeight = 60.0;
    [self.tableView registerClass:UniversalTableViewCell.class forCellReuseIdentifier:UniversalTableViewCell.reuseId];
	
	if (!self.tableView.tableHeaderView) self.tableView.tableHeaderView = [[UIView alloc] init];
	if (!self.tableView.tableFooterView) self.tableView.tableFooterView = [[UIView alloc] init];
}

#pragma mark - Table view data source

- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
    return 1;
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return self.cellNames.count;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    UniversalTableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:UniversalTableViewCell.reuseId];
    [cell updateWithPrimaryText:[self.cellNames objectAtIndexSafe:indexPath.row] secondaryText:nil];
    return cell;
}

#pragma mark - Table view delegate

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
	if (!indexPath) return;
	
	SearchSongsViewController *searchView = [[SearchSongsViewController alloc] initWithNibName:@"SearchSongsViewController" bundle:nil];
	NSString *type = [self.cellNames objectAtIndexSafe:indexPath.row];
	if ([type isEqualToString:@"Artists"]) {
		searchView.listOfArtists = [NSMutableArray arrayWithArray:self.listOfArtists];
		searchView.searchType = ISMSSearchSongsSearchType_Artists;
	} else if ([type isEqualToString:@"Albums"]) {
		searchView.listOfAlbums = [NSMutableArray arrayWithArray:self.listOfAlbums];
		searchView.searchType = ISMSSearchSongsSearchType_Albums;
	} else if ([type isEqualToString:@"Songs"]) {
		searchView.listOfSongs = [NSMutableArray arrayWithArray:self.listOfSongs];
		searchView.searchType = ISMSSearchSongsSearchType_Songs;
	}
	searchView.query = self.query;
	[self pushViewControllerCustom:searchView];
}

@end

File: Classes/View Controllers/Tabs/Home/SearchAllViewController.swift
//
//  SearchAllViewController.swift
//  iSub
//
//  Created by Benjamin Baron on 12/22/20.
//  Copyright © 2020 Ben Baron. All rights reserved.
//

import UIKit

@objc class SearchAllViewController: UIViewController {
    private var cellNames = [String]()
    
    private let tableView = UITableView()

    var folderArtists = [FolderArtist]()
    var folderAlbums = [FolderAlbum]()
    var songs = [Song]()
    var query = ""
    
    override func viewDidLoad() {
        super.viewDidLoad()
        if folderArtists.count > 0 { cellNames.append("Artists") }
        if folderAlbums.count > 0 { cellNames.append("Albums") }
        if songs.count > 0 { cellNames.append("Songs") }
        setupDefaultTableView(tableView)
    }
}
 
extension SearchAllViewController: UITableViewConfiguration {
    func numberOfSections(in tableView: UITableView) -> Int {
        return 1
    }
    
    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return cellNames.count
    }
    
    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell = tableView.dequeueUniversalCell()
        cell.update(primaryText: cellNames[indexPath.row], secondaryText: nil)
        return cell
    }
    
    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        let controller = SearchSongsViewController()
        switch cellNames[indexPath.row] {
        case "Artists":
            controller.folderArtists = folderArtists
            controller.searchType = .artists
        case "Albums":
            controller.folderAlbums = folderAlbums
            controller.searchType = .albums
        case "Songs":
            controller.songs = songs
            controller.searchType = .songs
        default: break
        }
        controller.query = query
        pushViewControllerCustom(controller)
    }
}
